coding-the-legislation/reforms.md

```diff
@@ -1,104 +1,94 @@
 
-The `reforms.make_reform` function does the job.
+    def function(household, period):
+        (...)
 
-Here is how to do a reform which does nothing, for the purpose of the example:
+class income_tax_reform(Reform):
+    name = u'Reform on income tax'
 
-```python
-def build_reform(tax_benefit_system):
-    Reform = reforms.make_reform(
-        key = 'empty',
-        name = u'Dummy empty reform',
-        reference = tax_benefit_system,
-        )
-    reform = Reform()
-    return reform
+    def apply(self):
+        self.update_variable(income_tax)
 ```
 
-> `reforms.make_reform` returns a `Reform` class that you can instantiate in `build_reform`.
-> You can modify this instance if needed, then return it.
+A `Reform` **must** define an `apply` method that describes all the modifications to be applied to the original tax benefit system to get the experimental one.
 
-The `Reform` class provides a decorator to declare formulas (`@Reform.formula`) and a function to declare input variables (`@Reform.input_variable`).
+> Note that the reference tax and benefit system won't be modified. The `apply` function will applied to a copy of the tax and benefit system.
 
-The reference `tax_benefit_system` won't be touched.
+All the [methods](https://openfisca.readthedocs.io/en/latest/tax-benefit-system.html) used to build a tax benefit system can also be used to reform it.
 
-Here is an example:
+A reform that modifies a formula (such as our `income_tax_reform` example) is called a *structural reform*. It redefines the way a variable is calculated.
 
-```python
-def build_reform(tax_benefit_system):
-    Reform = reforms.make_reform(
-        key = 'change_formula',
-        name = u'Dummy reform with changed formula',
-        reference = tax_benefit_system,
-        )
-
-    @Reform.formula
-    class charges_deduc(formulas.SimpleFormulaColumn):
-        label = u"Charge déductibles always returning 999"
-        reference = charges_deductibles.charges_deduc
-        definition_period = YEAR
-        
-        def function(self, simulation, period):
-            return self.zeros() + 999
-
-    Reform.input_variable(
-        column = columns.BoolCol,
-        entity_class = entities.Menages,
-        name ='parisien',
-        label = u"Résidant à Paris au moins 3 ans dans les 5 dernières années",
-        )
-
-    reform = Reform()
-    return reform
-```
 
-To change the JSON data structure of the legislation in the reform, you call the `reform.modify_legislation_json` method which takes as a parameter a callback.
-This callback is a function you write in which the legislation JSON will be modified.
-It takes in parameter a copy of the legislation_json of the reference tax and benefit system that you can modify (do not make the copy yourself) and return.
+### Parametric reforms
+
+Unlike *structural reforms*, *parametric reforms* do not directly modify any formula: they apply changes to the legislation parameters.
```

Comment:
“they apply changes to the legislation parameters“ → “they apply changes to legislation parameters only”